dask/array/creation.py

```diff
@@ -731,10 +731,10 @@ def offset_func(*args, **kwargs):
 def fromfunction(func, chunks="auto", shape=None, dtype=None, **kwargs):
     chunks = normalize_chunks(chunks, shape, dtype=dtype)
     name = "fromfunction-" + tokenize(func, chunks, shape, dtype, kwargs)
-    keys = list(product([name], *[range(len(bd)) for bd in chunks]))
+    keys = product([name], *[range(len(bd)) for bd in chunks])
     aggdims = [list(accumulate(add, (0,) + bd[:-1])) for bd in chunks]
-    offsets = list(product(*aggdims))
-    shapes = list(product(*chunks))
+    offsets = product(*aggdims)
```

Comment:
No need to materialize a list for `aggdims` here, can just create a generator and unpack that.